### ISSUE TITLE
[ELF] Drop duplicate rpath entries

### ISSUE THREAD

**Issue by soerengrunewald** [rui314/mold#873]
We have several libraries which we are utilizing via pkg-config. Every library provides its own `-Wl,-rpath` linker option. And this is absolutely reasonable, since every library can be used independently. The actual problem arises if they are used together, as this leads to some ridiculously long runpath's like this:
> 0x000000000000001d (RUNPATH)            Library runpath: [/opt/foo/bar/lib:/opt/foo/bar/lib:/opt/foo/bar/lib:/opt/foo/bar/lib:/opt/foo/bar/lib]

This can be easily reproduced by adding a couple more `-Wl,-rpath,/foo` to `test/elf/rpath.sh` and then run it `mold -run ./test/elf/rpath.sh`:
```diff
diff --git a/test/elf/rpath.sh b/test/elf/rpath.sh
index e2ec11e3..20afcb3f 100755
--- a/test/elf/rpath.sh
+++ b/test/elf/rpath.sh
@@ -8,8 +8,8 @@ main:
   nop
 EOF
 
-$CC -B. -o $t/exe $t/a.o \
-  -Wl,-rpath,/foo -Wl,-rpath,/bar -Wl,-R/no/such/directory -Wl,-R/
+mold --run $CC -B. -o $t/exe $t/a.o \
+  -Wl,-rpath,/foo -Wl,-rpath,/foo -Wl,-rpath,/foo -Wl,-rpath,/bar -Wl,-R/no/such/directory -Wl,-R/
 
 readelf --dynamic $t/exe | \
   grep -Fq 'Library runpath: [/foo:/bar:/no/such/directory:/]'
``` 
The default linker will remove the extra `/foo`'s while mold will keep them.

**Comment by soerengrunewald** [comment]
I have created a possible solution, but it is terrible https://github.com/soerengrunewald/mold/commit/ed85ce6e7dc32409ee5c73b3d7f66a4f1846ae7b and should not been used.

Also the *macho* implementation uses a `std::vector<std::string>` unlike the *elf* implementation, which uses a simple `std::string` to store the rpath elements.  I think it would be best to also use some container for the *elf* implementation.

Anyhow I have not found a suitable spot where one could generate the resulting rpath string (see elf/output-chunks.cc#L716 and elf/main.cc#L611). And one might wanna keep the original order in which the rpath's have been added, as it might be important to keep the intended search order. Is is also the reason for the strange use of `std:unordered_set` in my attempt.
